src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,107 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import * as clientComments from '../service/comment';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  createComment: (comment: Comment) => void
+  post: Post
+};
+
+export const NewCommentForm: React.FC<Props> = ({ createComment, post }) => {
+  const [nameText, setNameText] = useState('');
+  const [emailText, setEmailText] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [commentError, setCommentError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const reset = () => {
+    setNameText('');
+    setEmailText('');
+    setCommentText('');
+    setNameError(false);
+    setEmailError(false);
+    setCommentError(false);
+    setIsLoading(false);
+  };
+
+  const addComment = ({
+    postId, name, email, body,
+  }: Omit<Comment, 'id'>) => {
+    setIsLoading(true);
+
+    clientComments.createComment({
+      postId, name, email, body,
+    })
+      .then(createComment)
+      .finally(() => {
+        setCommentText('');
+        setIsLoading(false);
+      });
+  };
+
+  const changeNameText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (nameText.length === 0) {
+      setNameError(true);
+    }
+
+    setNameError(false);
+    setNameText(event.target.value);
+  };
+
+  const changeEmailText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (emailText.length === 0) {
+      setEmailError(true);
+    }
+
+    setEmailError(false);
+    setEmailText(event.target.value);
+  };
+
+  const changeCommentText = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    if (commentText.length === 0) {
+      setCommentError(true);
+    }
+
+    setCommentError(false);
+    setCommentText(event.target.value);
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form
+      data-cy="NewCommentForm"
+      onSubmit={(event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const isNameValid = nameText.trim();
+        const isEmailValid = emailText.trim();
+        const isBodyValid = commentText.trim();
+
+        if (!isNameValid) {
+          setNameError(true);
+        }
+
+        if (!isEmailValid) {
+          setEmailError(true);
+        }
+
+        if (!isBodyValid) {
+          setCommentError(true);
+        }
+
+        if (isNameValid && isEmailValid && isBodyValid) {
+          addComment({
+            postId: post.id,
+            name: nameText,
+            email: emailText,
+            body: commentText,
+          });
+        }
+      }}
```

Comment:
```suggestion
      onSubmit={(event: React.FormEvent<HTMLFormElement>) => {
          event.preventDefault();

          const isNameValid = nameText.trim();
          const isEmailValid = emailText.trim();
          const isBodyValid = commentText.trim();

          if (!isNameValid) {
            setNameError(true);
            return;
          }

          if (!isEmailValid) {
            setEmailError(true);
            return;
          }

          if (!isBodyValid) {
            setCommentError(true);
            return;
          }

          addComment({
            postId: post.id,
            name: nameText,
            email: emailText,
            body: commentText,
          });
        }}
```
also please move it to a separate function

Reply:
It doesn't work with return in each "if" block, because function return after first validation, but I need to make all of them and only after make something, and I can't delete if(isNameValid && isEmailValid && isBodyValid){...}